applications/serial_lte_modem/src/dfu/slm_at_dfu.c

```diff
@@ -307,8 +307,10 @@ static  int dfu_download_start(const char *host, const char *file, int sec_tag,
 	const char *file_buf_ptr = file_buf;
 	int err = -1;
 
+	int sec_tag_list[] = { sec_tag };
 	struct download_client_cfg config = {
-		.sec_tag = sec_tag,
+		.sec_tag_list = sec_tag_list,
+		.sec_tag_count = (sec_tag != -1) ? 1 : 0,
 		.frag_size_override = fragment_size,
 		.set_tls_hostname = (sec_tag != -1),
 	};
```

Comment:
Wov.. there seem to be a stack / object lifetime issue.

`struct download_client_cfg` is allocated from stack, but the Download client may refer to it from its internal thread.

While you are here, please move the configuration structure outside from stack allocation.

Reply:
Sorry, my bad.. the configuration structure seem to be copied. But the `sec_tag_list` must then have a lifetime as long as the download client is running. So at least the list needs to be global static.

Reply:
Lifetime issue should be fixed.